src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -89,6 +91,15 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+static int do_unmap();
+
+static void handle_signal(int signum)
+{
+  assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  ioctl(nbd, NBD_DISCONNECT);
```

Comment:
@trociny, I also find such kind of issue when I use unmap. Yeah，message is useful, not only in this signal handler, also for unmap function.

Just one question, what is the situation ioctl(nbd, NBD_DISCONNECT) may fail?